src/main/java/org/aksw/simba/squirrel/analyzer/impl/MimeTypeDetector.java

```diff
@@ -0,0 +1,43 @@
+package org.aksw.simba.squirrel.analyzer.impl;
+
+import org.aksw.simba.squirrel.analyzer.TypeDetector;
+import org.apache.jena.riot.Lang;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+public class MimeTypeDetector implements TypeDetector {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RDFAnalyzer.class);
+
+    public Lang detectMimeType(File data) {
+
+        Lang mimeType = null;
+        try {
+            FileInputStream inputStream = new FileInputStream(data.getAbsolutePath());
+            FiniteStateMachine machine1 = RdfAutomata.buildRDFStateMachine();
+            FiniteStateMachine machine2 = TurtleAutomata.buildTurtleStateMachine();
```

Comment:
The way you are creating the machines can be improved. Imagine we want to add a new machine. We would have to touch this method again and again. It would be better if you could define something like a `FiniteStateMachineFactory` which offers a `create()` method that gives you a new instance. The `MimeTypeDetector` could have a private list of these factories and simply get one instance from every factory when the `detectMimeType` method is called.